app/test/testMetadata.js

```diff
@@ -0,0 +1,192 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const fetchMock = require('fetch-mock');
+const {app} = require('../service');
+
+chai.use(chaiHttp);
+chai.should();
+
+const exampleMetadata = {
+  description: 'An example description',
+  favicon_url: 'http://www.example.com/rich-icon.png',
+  icon_url: 'http://www.example.com/rich-icon.png',
+  image_url: 'http://www.example.com/preview.png',
+  images: [
+    {
+      'url': 'http://www.example.com/preview.png',
+      'entropy': 1,
+      'height': 500,
+      'width': 500,
+    }
+  ],
+  original_url: 'http://www.example.com/good',
+  provider_url: 'http://www.example.com/good',
+  title: 'An Example Page',
+  url: 'http://www.example.com/good'
+};
+
+const goodExampleUrl = 'http://www.example.com/good';
+const badExampleUrl = 'http://www.example.com/bad';
+
+
+describe('Metadata API Tests', function() {
+  before(function() {
+    fetchMock.mock(
+      goodExampleUrl,
+      `
+        <html>
+          <head>
+            <title>${exampleMetadata.title}</title>
+            <meta name="description" content="${exampleMetadata.description}" />
+            <meta name="thumbnail" content="${exampleMetadata.images[0].url}" />
+            <link rel="icon" href="http://www.example.com/rich-icon.png" />
+          </head>
+          <body></body>
+        </html>
+      `
+    );
+  });
+
+  after(function() {
+    fetchMock.restore();
+  });
+
+  it('should raise 400 if missing JSON body', (done) => {
+    chai.request(app)
+      .post('/')
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
+
+  it('should raise 400 if missing urls parameter', (done) => {
+    chai.request(app)
+      .post('/')
+      .send(JSON.stringify({}))
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
+
+  it('should raise 400 if urls is empty', (done) => {
+    chai.request(app)
+      .post('/')
+      .send(JSON.stringify({urls: []}))
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
```

Comment:
Can you explain this one to me? The test passes (yay), but I'm not sure if it's valid (not yay).
Basically, I'm uncertain if we're getting a 400 because the content header is not set, or if the `urls[]` is empty.

This basic cURL action shows that a payload of `{"urls": []}` returns an `HTTP/1.1 200 OK` response:

```sh
$ curl -XPOST -i -H "content-type: application/json" -d '{"urls": []}' http://localhost:7001 | JSON

HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 22
ETag: W/"16-urTtGfwwfQX5N25qpNbXOg"
Date: Tue, 09 Aug 2016 23:23:11 GMT
Connection: keep-alive

{
  "error": "",
  "urls": {}
}
```